src/replayer.py

```diff
@@ -0,0 +1,306 @@
+
+    session = boto3.session.Session()
+    default_credentials = session.get_credentials().get_frozen_credentials()
+
+    loaded_event = json.loads(event)
+    try:
+        original_request = loaded_event["Body"]["originalRequest"]
+        original_response = loaded_event["Body"]["originalResponse"]
+    except Exception as e:
+        logger.error("Attempted to extract event items but was unable.")
+
+    nino = original_request.get("nino")
+    transaction_id = original_request.get("transaction_id")
+    from_date = original_request.get("from_date")
+    to_date = original_request.get("to_date")
+
+    actual_response = replay_original_request(
+        default_credentials, nino, transaction_id, from_date, to_date
+    )
+
+    decrypted_original_response = decrypt_response(original_response, original_request)
+    decrypted_actual_response = decrypt_response(actual_response, original_request)
+
+    if compare_responses(
+        decrypted_original_response, decrypted_actual_response, original_request
+    ):
+        logger.info('Final result", "status": "match')
+    else:
+        logger.info('Final result", "status": "miss')
+
+
+def replay_original_request(
+    default_credentials, nino, transaction_id, from_date, to_date
+):
+    auth = AWSRequestsAuth(
+        aws_access_key=default_credentials.access_key,
+        aws_secret_access_key=default_credentials.secret_key,
+        aws_token=default_credentials.token,
+        aws_host=f"{args.api_hostname}",
+        aws_region=f"{args.aws_region}",
+        aws_service="execute-api",
+    )
+
+    request_parameters = f"nino={nino}&transactionId={transaction_id}&fromDate={from_date}&toDo={to_date}"
+
+    headers = {
+        "Content-Type": "application/json",
+        "X-Amz-Date": datetime.datetime.utcnow().strftime("%Y%m%dT%H%M%SZ"),
+    }
+
+    logger.info(f'Requesting data from AWS API", "api_hostname": "{args.hostname}')
+    request = requests.post(
+        f"https://{args.api_hostname}/ucfs-claimant/v2/getAwardDetails",
+        data=request_parameters,
+        auth=auth,
+        headers=headers,
+    )
+
+    logger.info(
+        f'Received response from AWS API", "api_hostname": "{args.hostname}", "response_code": "{request.status_code}'
+    )
+
+    return json.load(StringIO(request.text))
+
+
+def decrypt_response(response: dict, request: dict) -> dict:
+    # Create a deep copy of the response to keep the function pure
+    response = response.copy()
+    session = boto3.session.Session(profile_name="decrypt", region_name=region)
+
+    client = session.client("kms")
+
+    for period in response.get("assessmentPeriod", []):
+        amount = period.get("amount")
+
+        key_id = amount.get("keyId")
+        take_home_pay = base64.urlsafe_b64decode(amount.get("takeHomePay"))
+        cipher_text_blob = base64.urlsafe_b64decode(amount.get("cipherTextBlob"))
+
+        kms_response = client.decrypt(CiphertextBlob=cipher_text_blob, KeyId=key_id)
```

Comment:
don't need key_id for KMS
```suggestion
        kms_response = client.decrypt(CiphertextBlob=cipher_text_blob)
```